models/utilities/default_model_mod.f90

```diff
@@ -283,6 +300,162 @@ subroutine pert_model_copies(state_ens_handle, ens_size, pert_amp, interf_provid
 
 end subroutine pert_model_copies
 
+!--------------------------------------------------------------------
+
+!> Reads in model_nml:model_state_variables and returns a
+!> state_var_type state_vars with nvars ; netcdf variable names ;
+!> qtys (kinds) ; clamp values ; updates
+!
+!> Verifies that the namelist was filled in correctly, and checks
+!> that there are valid entries for the dart_kind.
+
+subroutine get_state_variables_clamp(nml_state_vars, MAX_STATE_VARIABLES, use_clamping, state_vars)
+
+character(len=*),   intent(inout) :: nml_state_vars(:)
+integer,               intent(in) :: MAX_STATE_VARIABLES
+logical,               intent(in) :: use_clamping
+type(state_var_type), intent(out) :: state_vars
+
+character(len=NF90_MAX_NAME) :: netcdf_var_name, dart_qty_str, update
+character(len=256) :: string1, string2
+integer :: i, ivar
+
+if(.not. use_clamping) then
+   string1 = 'logical use_clamping set to false when values for clamping are required in this model nml'
+   call error_handler(E_ERR, 'get_state_variables_clamp', string1)
+endif
```

Comment:
I don't follow the logic of the use_clamping argument and error checking for it. 

call get_state_variables_clamp
or call get_state_variables

If you are overloading then the arguments should be different, so no need to check.

If you're overloading but also passing a logical for what the routine should do, it is a good time to step back and reassess whether overloading is a good thing to do here.